File: src/commonTest/kotlin/com/scoreo/infrastructure/InMemoryMatchDraftRepository.kt
package com.scoreo.infrastructure

import com.scoreo.domain.model.MatchDraft
import com.scoreo.domain.port.MatchDraftRepository

class InMemoryMatchDraftRepository : MatchDraftRepository {
    private var draft: MatchDraft? = null

    override fun save(draft: MatchDraft) {
        this.draft = draft
    }

    override fun load(): MatchDraft? = draft

    override fun clear() {
        draft = null
    }
}

File: src/infrastructure/testing/inMemoryMatchDraftRepository.ts
import type { MatchDraft } from '../../domain/model/matchDraft'
import type { MatchDraftRepository } from '../../domain/port/matchDraftRepository'

export class InMemoryMatchDraftRepository implements MatchDraftRepository {
  private draft: MatchDraft | undefined

  save(draft: MatchDraft): void {
    this.draft = draft
  }

  load(): MatchDraft | undefined {
    return this.draft
  }

  clear(): void {
    this.draft = undefined
  }
}
